marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -577,23 +562,68 @@ renderPartyPastActions stepNumber { tzOffset } state { inputs, interval, party }
       ]
 
-renderTimeout :: forall p a. Input -> Int -> Slot -> HTML p a
-renderTimeout { tzOffset } stepNumber timeoutSlot =
+renderTimeout :: forall p a. Input -> State -> Int -> TimeoutInfo -> HTML p a
+renderTimeout { tzOffset } state stepNumber timeoutInfo =
   let
-    timedOutDate =
-      maybe
-        "invalid date"
-        (\dt -> formatDate tzOffset dt <> " at " <> formatTime tzOffset dt)
-        (slotToDateTime timeoutSlot)
+    mTimeoutDateTime = slotToDateTime timeoutInfo.slot
+
+    timeoutDate = maybe "-" (formatDate tzOffset) mTimeoutDateTime
+
+    timeoutTime = maybe "-" (formatTime tzOffset) mTimeoutDateTime
+
+    header =
+      div
+        [ classNames [ "py-2", "px-3", "w-full", "flex", "justify-between" ] ]
+        [ div [ classNames [ "flex", "items-center", "text-xs" ] ]
+            [ icon Icon.Timer [ "mr-1" ]
+            , span [ classNames [ "font-semibold" ] ] [ text "Timed out" ]
+            ]
+        , div [ classNames [ "flex", "flex-col", "items-end", "text-xxs", "font-semibold" ] ]
+            [ span_ [ text timeoutDate ]
+            , span_ [ text $ timeoutTime <> " (" <> humanizeOffset tzOffset <> ")" ]
+            ]
+        ]
+
+    body =
+      div [ classNames [ "py-2", "px-3", "border-t", "border-gray" ] ]
+        $ renderMissingActions state timeoutInfo
   in
-    div [ classNames [ "flex", "flex-col", "items-center", "px-4" ] ]
-      -- NOTE: we use pt-16 instead of making the parent justify-center because in the design it's not actually
-      --       centered and it has more space above than below.
-      [ icon Icon.Timer [ "pb-2", "pt-16", "text-red", "text-big-icon" ]
-      , span [ classNames [ "font-semibold", "text-center", "text-sm" ] ]
-          [ text $ "Step " <> show (stepNumber + 1) <> " timed out on " <> timedOutDate ]
+    div [ classNames [ "mt-4", "bg-white", "rounded", "shadow-sm", "mx-4", "flex", "flex-col", "items-center" ] ]
+      [ header
+      , body
       ]
 
+renderMissingActions :: forall p a. State -> TimeoutInfo -> Array (HTML p a)
+renderMissingActions _ { missedActions: [] } =
+  [ div [ classNames [ "font-semibold", "text-xs", "leading-none" ] ] [ text "There were no tasks to complete at this step and the contract has timeouted as expected." ]
+  ]
+
+renderMissingActions state { missedActions } =
+  append
+    [ div [ classNames [ "font-semibold", "text-xs", "leading-none" ] ] [ text "The step timed out before the following actions could be made." ]
+    ]
+    (missedActions <#> uncurry (renderPartyMissingActions state))
+
+renderPartyMissingActions :: forall p a. State -> Party -> Array NamedAction -> HTML p a
+renderPartyMissingActions state party actions =
+  let
+    renderMissingAction (MakeChoice (ChoiceId name _) _ _) = span_ [ text "Make a choice for ", span [ classNames [ "font-semibold" ] ] [ text name ] ]
+
+    renderMissingAction (MakeDeposit _ _ token value) = span [] [ text $ "Make a deposit of " <> humanizeValue token value ]
```

Comment:
I just realised that this `value` could evolve over time, so looking at it in retrospective I am not sure what we are going to be able to say about it. This is not the case for prototype because all deposits are constant (I think), so we can go ahead for now, but when we go for `1.0` we'll need to think how to handle it in the general case.

Reply:
👍🏻 , yeah, right now the NamedAction have BigInteger for value but the language uses the `Value` type... we might need to settle the value with `evalValue` and store the result in the NamedAction or to change the type of NamedAction to store a `Value` and not sure how to present that to the user.